### DISCUSSION TITLE
Is there a way to watch the status of a CRD in a reconciler?

### DISCUSSION THREAD

**Discussion by Kiruthika-Jayabalan** [kubernetes-sigs/kubebuilder#3548] [Support]
```
// SetupWithManager sets up the controller with the Manager.
func (r *XyzReconciler) SetupWithManager(mgr ctrl.Manager) error {
	return ctrl.NewControllerManagedBy(mgr).
		For(&spacesv1alpha1.Profile{},
			builder.WithPredicates(predicate.Or(predicate.GenerationChangedPredicate{}))).
		Watches(&source.Kind{Type: &v1.ObjStatus{}},

		).
		Complete(r)
}
```

**Comment by camilamacedo86** [comment]
Please check the documentation: https://book.kubebuilder.io/reference/watching-resources

If you still with questions feel free to raise an issue for we improve the doc, also you can help out on it and looking for answers in the controller-runtime channel too. 

Closing this one for now, but feel free to re-open if you need/
